<?xml version="1.0"?>
<robot name="my_4dof_robot_box">

  <material name="grey">
    <color rgba="0.5 0.5 0.5 1.0"/>
  </material>
  <material name="blue">
    <color rgba="0.1 0.1 0.8 1.0"/>
  </material>

  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.10 0.10 0.02"/>
      </geometry>
      <material name="grey"/>
    </visual>
  </link>

  <link name="link_1">
    <visual>
      <geometry>
        <box size="0.04 0.04 0.10"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="joint_1" type="revolute">
    <parent link="base_link"/>
    <child link="link_1"/>
    <origin xyz="0 0 0.0814" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit lower="-1.54" upper="1.54" effort="1.08" velocity="7.48"/>
  </joint>

  <link name="link_2">
    <visual>
      <geometry>
        <box size="0.035 0.035 0.12"/>
      </geometry>
      <material name="grey"/>
    </visual>
  </link>
  <joint name="joint_2" type="revolute">
    <parent link="link_1"/>
    <child link="link_2"/>
    <origin xyz="-0.02 0 0.015" rpy="0 0 0"/>
    <axis xyz="0 -1 0"/>
    <limit lower="-1.54" upper="1.54" effort="2.63" velocity="7.48"/>
  </joint>

  <link name="link_3">
    <visual>
      <geometry>
        <box size="0.03 0.03 0.15"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="joint_3" type="revolute">
    <parent link="link_2"/>
    <child link="link_3"/>
    <origin xyz="-0.0021 0.00 0.1035" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit lower="-1.54" upper="1.54" effort="1.08" velocity="7.48"/>
  </joint>

  <link name="link_4">
    <visual>
      <geometry>
        <box size="0.025 0.025 0.13"/>
      </geometry>
      <material name="grey"/>
    </visual>
  </link>
  <joint name="joint_4" type="revolute">
    <parent link="link_3"/>
    <child link="link_4"/>
    <origin xyz="-0.002 0 0.1275" rpy="0 0 0"/>
    <axis xyz="0 -1 0"/>
    <limit lower="-1.54" upper="1.54" effort="1.08" velocity="7.48"/>
  </joint>

  <link name="tool_tip">
    <visual>
      <geometry>
        <box size="0.02 0.02 0.04"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="joint_tool" type="fixed">
    <parent link="link_4"/>
    <child link="tool_tip"/>
    <origin xyz="0 0 0.095" rpy="0 0.175 0"/>
  </joint>

</robot>